src/nbl/video/IDescriptorPool.cpp

```diff
@@ -0,0 +1,67 @@
+#include "nbl/video/IDescriptorPool.h"
+#include "nbl/video/IGPUDescriptorSetLayout.h"
+
+namespace nbl::video
+{
+
+bool IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
+{
+    core::vector<SDescriptorOffsets> descriptorOffsets(count);
+
+    for (uint32_t i = 0u; i < count; ++i)
+    {
+        if (!isCompatibleDevicewise(layouts[i]))
+            return false;
+        
+        descriptorOffsets[i] = allocateDescriptorOffsets(layouts[i]);
+    }
+
+    return createDescriptorSets_impl(count, layouts, descriptorOffsets.data(), output);
+}
+
+bool IDescriptorPool::reset()
+{
+    const bool allowsFreeing = m_creationParameters.flags.hasFlags(ECF_FREE_DESCRIPTOR_SET_BIT);
+    for (uint32_t t = 0; t < static_cast<uint32_t>(asset::IDescriptor::E_TYPE::ET_COUNT); ++t)
+    {
+        if (!m_descriptorAllocators[t])
+            continue;
+
+        // TODO(achal): I should call the dtors on descriptors only after calling vkResetDescriptorPool (in reset_impl)
+        const uint32_t allocatedCount = m_descriptorAllocators[t]->getAllocatedDescriptorCount(allowsFreeing);
+        std::destroy_n(getDescriptorStorage(static_cast<asset::IDescriptor::E_TYPE>(t)), allocatedCount);
```

Comment:
We don't need `std::destroy_n` at all

Reply:
we do cause storage is trivialized for now

Reply:
we can keep storage trivialized, so they're needed